core/camel-core-model/src/main/java/org/apache/camel/model/ThrottleDefinition.java

```diff
@@ -89,35 +120,132 @@ public String getShortName() {
+        return this;
+    }
+
+    /**
+     * Sets the maximum number of requests
+     *
+     * @param  maximumConcurrentRequests the maximum number of requests (according to the mode in use - either
+     *                                   concurrent or by time period)
      * @return                           the builder
      */
-    public ThrottleDefinition maximumConcurrentRequests(long maximumConcurrentRequests) {
+    public ThrottleDefinition maximumRequests(long maximumConcurrentRequests) {
         setExpression(
                 ExpressionNodeHelper.toExpressionDefinition(ExpressionBuilder.constantExpression(maximumConcurrentRequests)));
         return this;
     }
 
     /**
-     * Sets the number of concurrent requests
+     * Sets the maximum number of requests
      *
-     * @param  maximumConcurrentRequests the maximum number of concurrent requests
+     * @param  maximumConcurrentRequests the maximum number of requests (according to the mode in use - either
+     *                                   concurrent or by time period)
      * @return                           the builder
      */
-    public ThrottleDefinition maximumConcurrentRequests(String maximumConcurrentRequests) {
+    public ThrottleDefinition maximumRequests(String maximumConcurrentRequests) {
         setExpression(
-                ExpressionNodeHelper.toExpressionDefinition(ExpressionBuilder.simpleExpression(maximumConcurrentRequests)));
+                ExpressionNodeHelper.toExpressionDefinition(ExpressionBuilder.constantExpression(maximumConcurrentRequests)));
         return this;
     }
 
+    /**
+     * Sets the maximum number of concurrent requests
+     *
+     * @param      maximumConcurrentRequests the maximum number of concurrent requests
+     * @deprecated                           Use {@link #maximumRequests(long)}
+     * @return                               the builder
+     */
+    @Deprecated(since = "4.4.0")
+    public ThrottleDefinition maximumConcurrentRequests(long maximumConcurrentRequests) {
+        if (ThrottlingMode.toMode(mode) == ThrottlingMode.ConcurrentRequests) {
+            setExpression(
+                    ExpressionNodeHelper
+                            .toExpressionDefinition(ExpressionBuilder.constantExpression(maximumConcurrentRequests)));
+            return this;
+        } else {
+            throw new IllegalArgumentException(
+                    "Maximum concurrent requests can only be set when using concurrent requests mode");
+        }
+    }
+
+    /**
+     * Sets the number of concurrent requests
+     *
+     * @param      maximumConcurrentRequests the maximum number of concurrent requests
+     * @deprecated                           Use {@link #maximumRequests(long)}
+     * @return                               the builder
+     */
+    @Deprecated(since = "4.4.0")
+    public ThrottleDefinition maximumConcurrentRequests(String maximumConcurrentRequests) {
```

Comment:
The new throttler code was added in 4.3 (non LTS) so IMHO its fine to remove/change the code as upgrade from LTS to LTS is what matters.